File: Lab1/Temperature/Kelvin.cs
namespace Temperature;

class Kelvin : Temperature
{
    public Kelvin(double kelvin) : base(kelvin)
    {
        if (kelvin < 0) throw new ArgumentException("Absolute zero on Celsius is 0K.");
    }

    public Celsius ToCelsius()
    {
        return new Celsius(this.value - 273.15);
    }

    public Fahrenheit ToFahrenheit()
    {
        return ToCelsius().ToFahrenheit();
    }
}

File: Lab1/src/Temperature/Kelvin.java
package SoftwareEngineering.Lab1.Temperature;

public class Kelvin extends BaseTemperature {
    public Kelvin(double temperature) {
        // TODO Absolute zero on Celsius is 0K
        super(temperature);
    }

    public Celsius toCelsius() {
        return new Celsius(temperature - 273.15);
    }

    public Fahrenheit toFahrenheit() {
        return toCelsius().toFahrenheit();
    }

    public Kelvin toKelvin() {
        return new Kelvin(temperature);
    }
}